kernel/src/hil/spi.rs

```diff
@@ -40,6 +40,42 @@ pub enum ClockPhase {
     SampleTrailing,
 }
 
+pub mod util {
+    #[derive(Copy, Clone)]
+    pub enum ChipSelectActivePolarity {
+        ActiveLow,
+        ActiveHigh,
+    }
+
+    #[derive(Copy, Clone)]
+    pub struct ChipSelect<P> {
+        pub pin: P,
+        pub polarity: ChipSelectActivePolarity,
+    }
```

Comment:
Do you have an example of what this looks like if `pin` is not a GPIO pin, but rather some kind of opaque number?

Reply:
It would rely on there being some HW register or something that sets the polarity, but:

```rust
impl SpiMaster<...> for MySpi {
    type ChipSelect = ChipSelect<u8>;

    ...

    fn specify_chip_select(&self, cs: Self::ChipSelect) {
        self.registers.csid.set(cs.pin);
        self.registers.cspolarity.set(match self.polarity {
              Polarity::ActiveLow => regs::CSPOL::LOW,
              Polarity::ActiveHigh => regs::CSPOL::HIGH,
        });
    }
}
```